extensions/in-memory/policy-store-memory/src/main/java/org/eclipse/dataspaceconnector/policy/store/memory/InMemoryPolicyStore.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2020-2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.policy.store.memory;
+
+import org.eclipse.dataspaceconnector.common.concurrency.LockManager;
+import org.eclipse.dataspaceconnector.common.reflection.ReflectionUtil;
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.contract.policy.store.PolicyStore;
+import org.eclipse.dataspaceconnector.spi.persistence.EdcPersistenceException;
+import org.eclipse.dataspaceconnector.spi.query.BaseCriterionToPredicateConverter;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.query.SortOrder;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static org.eclipse.dataspaceconnector.common.reflection.ReflectionUtil.propertyComparator;
+
+/**
+ * An in-memory, threadsafe policy store.
+ * This implementation is intended for testing purposes only.
+ */
+public class InMemoryPolicyStore implements PolicyStore {
+
+    private final LockManager lockManager;
+    private final Map<String, Policy> policiesById =  new HashMap<>();
+
+    public InMemoryPolicyStore(LockManager lockManager) {
+        this.lockManager = lockManager;
+    }
+
+    public InMemoryPolicyStore() {
+        this.lockManager = new LockManager(new ReentrantReadWriteLock());
```

Comment:
nit: lock with fairness. 
```suggestion
        this.lockManager = new LockManager(new ReentrantReadWriteLock(true));
```

Reply:
changed